src/features/offline-checks/model/services/MediaLookupService.ts

```diff
@@ -0,0 +1,87 @@
+import { LookupMediaInput } from '@app/abstract/lib';
+import { ActivityDetails, ActivityModel } from '@app/entities/activity';
+import { ActivityResponse, AppletDetailsResponse } from '@app/shared/api';
+import {
+  getActivityDetailsKey,
+  getAppletDetailsKey,
+  getDataFromQuery,
+} from '@app/shared/lib';
+
+const createMediaLookupService = () => {
+  const lookupInMarkdown = (message: string): boolean => {
+    const videoFound = message.includes('<video');
+    const audioFound = message.includes('<audio');
+    const youtubeFound = message.includes('youtu');
+
+    return videoFound || audioFound || youtubeFound;
+  };
+
+  const lookupInActivity = (activity: ActivityDetails): boolean => {
+    for (let item of activity.items) {
+      if (lookupInMarkdown(item.question)) {
+        return true;
+      }
+
+      if (item.inputType === 'AudioPlayer') {
+        return true;
+      }
+    }
+    return false;
+  };
+
+  const lookup = ({
+    appletId,
+    entityId,
+    entityType,
+    queryClient,
+  }: LookupMediaInput): boolean => {
+    const activitiesToLookup: ActivityDetails[] = [];
+
+    const addActivity = (id: string) => {
+      const activityResponse = getDataFromQuery<ActivityResponse>(
+        getActivityDetailsKey(id),
+        queryClient,
+      );
+
+      const activity: ActivityDetails = ActivityModel.mapToActivity(
+        activityResponse!.result,
+      );
+      activitiesToLookup.push(activity);
+    };
+
+    if (entityType === 'regular') {
+      addActivity(entityId);
+    }
+
+    if (entityType === 'flow') {
+      const appletDetailsResponse = getDataFromQuery<AppletDetailsResponse>(
+        getAppletDetailsKey(appletId),
+        queryClient,
+      );
+
+      const flowDto = appletDetailsResponse?.result.activityFlows.find(
+        x => x.id === entityId,
+      );
+
+      const activityIds: string[] = flowDto!.activityIds;
+
+      for (let activityId of activityIds) {
+        addActivity(activityId);
+      }
+    }
+
+    for (let activity of activitiesToLookup) {
+      if (lookupInActivity(activity)) {
+        return true;
+      }
+    }
+
+    return false;
+  };
+
+  return {
+    lookup,
+  };
+};
+
+export default createMediaLookupService();
```

Comment:
I suggest moving the services to the activity entity. A feature implies user interactions with the system, and here in this case we have an internal business logic that needs no user interaction.